WcaOnRails/app/models/abstract_cached_model.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+class AbstractCachedModel < ActiveRecord::Base
```

Comment:
Re-reading my sentence it was not explicit but what I really meant is:

> However I would not like to duplicate code easy to read in the concrete class just because the code in the base class is hard to read

:p
I specifically don't want to implement the same simple `cfind` in all the concrete classes when they could share a more "difficult" `cfind` in their base class, for the exact same result.

I think your proposal is almost what I do, and would be doable in the base class using something like this:
```ruby
  def self.cfind(id)
    unless class_variable_defined?(:@@cache)
      class_variable_set(:@@cache, {})
    end
    unless class_variable_get(:@@cache).include?(id)
      class_variable_get(:@@cache)[id] = find(id)
    end
    class_variable_get(:@@cache)[id]
  end
```
Since all class methods in ruby are virtual, the actual `class_variable_set` or `_get` called is the one from the concrete class, which makes possible to set class variables in the concrete class from the base class, and to do what your propose from the base class.
(This is this non-obvious behavior I described in my updated comment in this class)

We also have multiple calls to `Model.all` which we want to cache too: do you have a solution to catch new records in these?
If we want to catch new records in `cfind` I'm not sure we can have one single cache object, because assuming `cfind` gets called before `all`, then `all` has no way to know the cache object is complete.

So according to my understanding we have two choices:
- share the cache object, and we fill it on the first call to either `all` or `cfind` (we don't catch new records).
- we keep two different caches for both `cfind` and `all`, we catch new records in `cfind` but I don't know yet how to catch them in `all`

This is regardless of where we set the concrete class variable from (base or concrete).